lib/hls/hls_parser.js

```diff
@@ -3457,6 +3457,12 @@ shaka.hls.HlsParser = class {
     if (this.isLive_() && mediaSequenceToStartTime.has(position)) {
       firstStartTime = mediaSequenceToStartTime.get(position);
     }
+    if (firstStartTime === 0 &&
```

Comment:
I'd add a comment saying this is for recovering from disconnects.

Reply:
Done!